Source/PlasticSourceControl/Private/PlasticSourceControlOperations.cpp

```diff
@@ -192,42 +192,49 @@ bool FPlasticCheckInWorker::Execute(FPlasticSourceControlCommand& InCommand)
-			FPlasticSourceControlModule& PlasticSourceControl = FModuleManager::GetModuleChecked<FPlasticSourceControlModule>("PlasticSourceControl");
-			FPlasticSourceControlProvider& Provider = PlasticSourceControl.GetProvider();
-
-			TArray<TSharedRef<ISourceControlState, ESPMode::ThreadSafe>> LocalStates;
-			Provider.GetState(InCommand.Files, LocalStates, EStateCacheUsage::Use);
-			for (const auto& State : LocalStates)
+			// Detect special case for a partial checkout (CS:-1 in Gluon mode)!
+			if (-1 != InCommand.ChangesetNumber)
+			{
+				Parameters.Add(TEXT("--all")); // Also files Changed (not CheckedOut) and Moved/Deleted Locally
+			//  NOTE: --update added as #23 but removed as #32 because most assets are locked by the Unreal Editor
+			//	Parameters.Add(TEXT("--update")); // Processes the update-merge automatically if it eventually happens.
+				InCommand.bCommandSuccessful = PlasticSourceControlUtils::RunCommand(TEXT("checkin"), Parameters, InCommand.Files, InCommand.Concurrency, InCommand.InfoMessages, InCommand.ErrorMessages);
+			}
+			else
+			{
+				Parameters.Add(TEXT("--applychanged")); // Also files Changed (not CheckedOut) and Moved/Deleted Locally
+				InCommand.bCommandSuccessful = PlasticSourceControlUtils::RunCommand(TEXT("partial checkin"), Parameters, InCommand.Files, InCommand.Concurrency, InCommand.InfoMessages, InCommand.ErrorMessages);
+			}
+			if (InCommand.bCommandSuccessful)
 			{
-				if (State->IsDeleted())
+				// Remove any deleted files from status cache
+				FPlasticSourceControlModule& PlasticSourceControl = FModuleManager::GetModuleChecked<FPlasticSourceControlModule>("PlasticSourceControl");
+				FPlasticSourceControlProvider& Provider = PlasticSourceControl.GetProvider();
+
+				TArray<TSharedRef<ISourceControlState, ESPMode::ThreadSafe>> LocalStates;
+				Provider.GetState(InCommand.Files, LocalStates, EStateCacheUsage::Use);
+				for (const auto& State : LocalStates)
 				{
-					Provider.RemoveFileFromCache(State->GetFilename());
+					if (State->IsDeleted())
+					{
+						Provider.RemoveFileFromCache(State->GetFilename());
+					}
 				}
-			}
 
-			Operation->SetSuccessMessage(ParseCheckInResults(InCommand.InfoMessages));
-			UE_LOG(LogSourceControl, Log, TEXT("CheckIn successful"));
+				Operation->SetSuccessMessage(ParseCheckInResults(InCommand.InfoMessages));
+				UE_LOG(LogSourceControl, Log, TEXT("CheckIn successful"));
+			}
 		}
-	}
 
-	// now update the status of our files
-	PlasticSourceControlUtils::RunUpdateStatus(InCommand.Files, false, InCommand.Concurrency, InCommand.ErrorMessages, States, InCommand.ChangesetNumber, InCommand.BranchName);
+		// now update the status of our files
+		PlasticSourceControlUtils::RunUpdateStatus(InCommand.Files, false, InCommand.Concurrency, InCommand.ErrorMessages, States, InCommand.ChangesetNumber, InCommand.BranchName);
+	}
+	else
+	{
+		UE_LOG(LogSourceControl, Warning, TEXT("Checkin operation without files"));
```

Comment:
If you look at other logs, it's not universal but I tend to use either the name of the operation or the name of the function at the start of the log like above
`"CheckIn: %d file(s) Description: '%s'"`
so in this case perhaps more something like this to be more uniform
`"CheckIn: no file"`